src/makeCalculator.js

```diff
@@ -37,7 +37,36 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const calculator = {
+    result: 0,
+    add(number, object = this) {
+      object.result += number;
+    },
+    subtract(number, object = this) {
+      object.add(-number);
+    },
+    multiply(number, object = this) {
+      object.result *= number;
+    },
+    divide(number, object = this) {
+      if (number === 0) {
+        throw new Error('divide by 0');
+      }
+      object.result /= number;
+    },
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+    operate(operation, number) {
```

Comment:
Instead of passing context as an argument, you could use `operation.call(this, number);` when invoking methods like `add` inside `operation`
This gives context to the methods called by `operation` and lets them use `this` directly. We are not taught `.call` or other object methods before this task, so I'm letting this solution pass.